python/desc/imsim/flats.py

```diff
@@ -0,0 +1,90 @@
+"""
+imSim module to make flats.
+"""
+import galsim
+import desc.imsim
+
+__all__ = ['make_flat']
+
+
+def make_flat(gs_det, counts_per_iter, niter, rng, buf=2, logger=None,
+              wcs=None):
+    """
+    Create a flat by successively adding lower level flat sky images.
+    This is based on
+    https://github.com/GalSim-developers/GalSim/blob/releases/2.0/devel/lsst/treering_flat.py
+
+    Full LSST CCDs are assembled one amp at a time to limit the memory
+    used by the galsim.SiliconSensor model.
+
+    Parameters
+    ----------
+    gs_det: GalSimDetector
+        The detector in the LSST focalplane to use.  This object
+        contains the CCD pixel geometry, WCS, and treering info.
+    counts_per_iter: int
+        Roughly the number of electrons/pixel to add at each iteration.
+    niter: int
+        Number of iterations. Final flat will have niter*counts_per_iter
+        electrons/pixel.
+    rng: galsim.BaseDeviate
+        Random number generator.
+    buf: int [2]
+        Pixel buffer around each to account for charge redistribution
+        across pixel boundaries.
+    logger: logging.Logger [None]
+        Logging object. If None, then a default logger will be used.
+    wcs: galsim.fitswcs.GSFitsWCS [None]
```

Comment:
Any `galsim.WCS` will work here.  The GSFitsWCS is a particular subclass that I used in my test.  But for instance, you use a PixelScale in one of your tests, which is another subclass of WCS.